pokemon-showdown

```diff
@@ -33,14 +33,16 @@ try {
 } catch (e) {}
 
 if (config && config.golang) {
-	if (!process.env.GOPATH) {
-		console.log('The GOPATH environment variable is not set! It is required in order to run the server using Go.');
-		process.exit(0);
-	}
-	if (!process.env.GOROOT) {
-		console.log('The GOROOT environment variable is not set! It is required in order to run the server using Go.');
-		process.exit(0);
-	}
+	// GOPATH and GOROOT are optional to a degree, but we need them in order
+	// to be able to handle Go dependencies. Since Go only cares about the
+	// first path in the list, so will we.
+	const GOPATH = child_process.execSync('go env GOPATH', {stdio: null, encoding: 'utf8'})
+		.trim().split(path.delimiter)[0];
+	process.env.GOPATH = GOPATH;
+
+	let GOROOT = child_process.execSync('go env GOROOT', {stdio: null, encoding: 'utf8'})
```

Comment:
I need `GOPATH` to spawn the Go child process, since it's located in `$GOPATH/bin/sockets`. and spawned from `sockets.js` instead of the launch script. Retrieving the value again isn't very taxing. I'll get rid of `GOPATH` if I find nothing actually needs it while I'm at it